src/gaxios.ts

```diff
@@ -144,10 +144,15 @@ export class Gaxios {
       }
       return translatedResponse;
     } catch (e) {
-      const err =
-        e instanceof GaxiosError
-          ? e
-          : new GaxiosError((e as Error).message, opts, undefined, e as Error);
+      let err: GaxiosError;
+
+      if (e instanceof GaxiosError) {
+        err = e;
+      } else if (e instanceof Error) {
+        err = new GaxiosError(e.message, opts, undefined, e);
```

Comment:
(just for my curiosity) - why wouldn't we contstruct the error with `e.cause` vs. `e.message`?

Reply:
It allows `GaxiosError` to optionally have a root cause (say, the error is simple or purely on the gaxios level).